project/dexter-metrics/src/test/com/samsung/sec/dexter/metrics/CodeMetricsGeneratorTest.java

```diff
@@ -0,0 +1,166 @@
+/**
+ * Copyright (c) 2017 Samsung Electronics, Inc.,
+ * All rights reserved.
+ * 
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * 
+ * * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ * 
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * 
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.samsung.sec.dexter.metrics;
+
+import static org.junit.Assert.assertTrue;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import com.puppycrawl.tools.checkstyle.api.AuditEvent;
+import com.puppycrawl.tools.checkstyle.api.CheckstyleException;
+import com.samsung.sec.dexter.core.config.DexterConfig;
+import com.samsung.sec.dexter.core.config.DexterConfig.LANGUAGE;
+import com.samsung.sec.dexter.core.metrics.CodeMetrics;
+import com.samsung.sec.dexter.core.metrics.FunctionMetrics;
+
+public class CodeMetricsGeneratorTest {
+	CodeMetricsGenerator tester = new CodeMetricsGenerator();
+	DexterConfig.LANGUAGE language = null;
+	String filePath = null;
+	CodeMetrics codeMetrics = null;
+	FunctionMetrics functionMetrics = null;
+	List<String> functionList = null;
+
+	String TestFilePath = ".\\src\\test\\com\\samsung\\sec\\dexter\\metrics\\TestFile_For_CodeMetricsGeneratorTest.txt";
+	String TestDirectoryPath = ".\\src\\test\\com\\samsung\\sec\\dexter\\metrics\\TestDirectory_For_CodeMetricsGeneratorTest";
```

Comment:
It would be better to move all the test sample files out of the 'src' folder. (eg. 'sample')
The 'src' folder should contains only the source codes to compile.